external-crates/move/move-compiler/src/typing/translate.rs

```diff
@@ -57,6 +57,10 @@ fn module(
 ) -> T::ModuleDefinition {
     assert!(context.current_script_constants.is_none());
     context.current_module = Some(ident);
+    // reset called functions and packed types sets so that they can be populated with values from
+    // this module only
+    context.called_fns.clear();
+    context.packed_types.clear();
```

Comment:
Maybe I'm just a bit crazy and open to discussion here, but I generally like to make sure things like this are already empty before entering the scope.

So in the style of the rest of the compiler, I would have 
```
assert!(context.called_fns.is_empty())`
```
And then the function has to manually clean them out below. 

I do this because I have been bitten in the past of silent failures with things like `.clear()`, but open to discussion here :)